tests/test.js

```diff
@@ -438,4 +438,173 @@ test('NbsDecoder.getPackets() returns the last packet of each type when given a
   });
 });
 
+const multiTypeNextTimestampArray = [
+  { type: pongType, subtype: 0 },
+  { type: pingType, subtype: 0 },
+  { type: pangType, subtype: 100 },
+  { type: pangType, subtype: 200 },
+];
+
+test('NbsDecoder.nextTimestamp() outputs the exact timestamp as input, given steps = 0', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  // const new_timestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 0);
+  const new_timestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 0);
+
+  assert.equal(
+    tsToBigInt(new_timestamp),
+    tsToBigInt(start),
+    'With steps = 0, input start timestamp should be equal to the returned timestamp for single type'
+  );
+});
+
+test('NbsDecoder.nextTimestamp() Stepping forward by one timestamp returns a timestamp which is greater than input.', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 1);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, multiTypeNextTimestampArray, 1);
+
+  assert.ok(
+    tsToBigInt(singleTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) > tsToBigInt(start),
+    'The new return Nbs timestamp is greater than the input timestamp'
+  );
+});
+
+test(`NbsDecoder.nextTimestamp() Stepping forward by five returns the expected timestamp for single and multi type`, () => {
```

Comment:
Tests should try and just test one thing and this more or less is but the description makes it sound like multiple tests
```suggestion
test(`NbsDecoder.nextTimestamp() Stepping forward by five returns the expected timestamp`, () => {
```

Reply:
Renamed.